src/execution/sql/inserter.cpp

```diff
@@ -0,0 +1,64 @@
+#include "execution/sql/inserter.h"
+#include <algorithm>
+
+namespace terrier::execution::sql {
+
+Inserter::Inserter(exec::ExecutionContext *exec_ctx, catalog::table_oid_t table_oid)
+    : table_oid_{table_oid}, exec_ctx_{exec_ctx} {
+  table_ = exec_ctx->GetAccessor()->GetTable(table_oid);
+  auto columns = exec_ctx_->GetAccessor()->GetSchema(table_oid_).GetColumns();
+  for (const auto &col : columns) {
+    col_oids_.push_back(col.Oid());
+  }
+
+  // getting index pr size
+  max_pr_size_ = 0;
+  auto index_oids = exec_ctx->GetAccessor()->GetIndexOids(table_oid_);
+  for (auto index_oid : index_oids) {
+    max_pr_size_ = std::max(max_pr_size_, GetIndex(index_oid)->GetProjectedRowInitializer().ProjectedRowSize());
+  }
+
+  index_pr_buffer_ = exec_ctx->GetMemoryPool()->AllocateAligned(max_pr_size_, sizeof(uint64_t), true);
+}
+
+Inserter::~Inserter() { exec_ctx_->GetMemoryPool()->Deallocate(index_pr_buffer_, max_pr_size_); }
+
+storage::ProjectedRow *Inserter::GetTablePR() {
+  // We need all the columns
+  storage::ProjectedRowInitializer pri = table_->InitializerForProjectedRow(col_oids_);
```

Comment:
Can we stash this in the Inserter? This is very expensive to create for every PR.